wormhole-connect/src/utils/wallet/solana.ts

```diff
@@ -44,6 +45,46 @@ import { VersionedTransaction } from '@solana/web3.js';
 const getWalletName = (wallet: Wallet) =>
   wallet.getName().toLowerCase().replaceAll('wallet', '').trim();
 
+// Checks response logs for known errors.
+// Returns when the first error is encountered.
+function checkKnownSimulationError(
+  response: SimulatedTransactionResponse,
+): boolean {
+  // This error occur when the blockhash included in a transaction is not deemed to be valid
+  // when a validator processes a transaction. We can retry the simulation to get a valid blockhash.
+  if (response.err === 'BlockhashNotFound') {
+    console.info('Blockhash not found during simulation. Trying again.');
+    return true;
+  }
+
+  // In some cases which aren't deterministic, like a slippage error, we can retry the
+  // simulation a few times to get a successful response.
+  if (response.logs) {
+    for (const line of response.logs) {
+      if (line.includes('SlippageToleranceExceeded')) {
+        console.info('Slippage failure during simulation. Trying again.');
+        return true;
+      }
+    }
+  }
+
+  // In this case a require_gte expression was violated during a Swap instruction.
+  // We can retry the simulation to get a successful response.
+  if (response.logs) {
+    for (const line of response.logs) {
+      if (line.includes('RequireGteViolated')) {
```

Comment:
Could we move this check into the block above here (since we're already iterating through the logs there), or do you prefer this to happen in a second pass of iterating through the logs?

Reply:
Do you think there is any possibility of having more than one of these errors in given logs? I was thinking maybe we should not bail out when see and error and aggregate all of them in the console.info?